io-engine/src/grpc/v1/pool.rs

```diff
@@ -337,38 +295,58 @@ impl PoolRpc for PoolService {
             GrpcClientContext::new(&request, function_name!()),
             async move {
                 let args = request.into_inner();
-                let pool_type = match &args.pooltype {
-                    Some(pool_type) => pool_type.value,
-                    None => PoolType::Lvs as i32,
+
+                // todo: what is the intent here when None, to only return pools
+                //  of Lvs?
+                // todo: Also, what todo when we hit an error listing any of the
+                //  types? Or should we have separate lists per type?
+                let pool_type = args.pooltype.as_ref().map(|v| v.value);
+                let pool_type = match pool_type {
+                    None => None,
+                    Some(pool_type) => {
+                        Some(PoolType::try_from(pool_type).map_err(|_| {
+                            Status::invalid_argument("Unknown pool type")
+                        })?)
+                    }
                 };
-                if pool_type != PoolType::Lvs as i32 {
-                    return Err(tonic::Status::invalid_argument(
-                        "Only pools of Lvs pool type are supported",
-                    ));
+
+                let lvm = matches!(pool_type, Some(PoolType::Lvm));
+                let lvs = matches!(pool_type, None | Some(PoolType::Lvs));
+
+                let mut pools = vec![];
+                if lvm {
+                    pools.extend(
+                        match LvmSvc::new().list_svc_pools(&args).await {
+                            Ok(pools) => Ok(pools),
+                            Err(mut status) => {
+                                status.metadata_mut().insert(
+                                    "lvm",
+                                    tonic::metadata::MetadataValue::from(0),
```

Comment:
Hmm maybe we should separate list by types?

Reply:
How do we use this metadata on caller side?

Reply:
We can separate, caller can just select the ones it wants.
In case caller selects all and an error is seen, caller side can check this metadata.